### PR TITLE
Add merge function

### PR DESCRIPTION
**Pull request by thedufer**

### CODE REVIEW

**Comment by thedufer** [comment]
Is this more or less how you would've implemented this?

**Comment by ianthehenry** [comment]
@thedufer Merge should always create a discrete signal -- it doesn't make sense to have a continuous merge. When you first create it, what's its value? The value of the last thing in the array, basically ignoring the other three. It makes sense to be *able* to merge continuous signals, to have a discrete signal-of-when-any-of-them-change, but I don't think it makes sense for that result to itself be continuous.

**Comment by thedufer** [comment]
Okay, I see what you're saying.  I was basing it off `combine`, but that wasn't a great analogy.

File: src/functions/merge.coffee [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Signal = require '../signal'
+
+module.exports = (signals...) ->
+  new Signal (send) ->
+    for signal in signals
+      signal.subscribe (val) ->
```
Comment (ianthehenry) [inline src/functions/merge.coffee]:
```
signal.subscribe(send)
```

Really,

```
signals.forEach f.method('subscribe', send)
```

But Hearsay doesn't want to depend on effing.

**Comment by ianthehenry** [comment]
(I'll fix it)

**Pull request merged**